packages/react/src/components/MultiSelect/MultiSelect.tsx

```diff
@@ -373,14 +373,29 @@ const MultiSelect = React.forwardRef(
       selectedItems: selected,
     });
 
+    // Filter out items with an object having undefined values
+    const filteredItems = items.filter((item) => {
+      if (typeof item === 'object' && item !== null) {
+        for (const key in item) {
+          if (
+            Object.prototype.hasOwnProperty.call(item, key) &&
```

Comment:
`hasOwn()` is a bit more intuitive and should be functionally the same I think

```suggestion
            Object.hasOwn(item, key) &&
```